File: Implementation/ClimbingTheLeaderboard.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the climbingLeaderboard function below.
function climbingLeaderboard(al1, al2) {
    al1 = [...new Set(al1)];
    console.log(al1);
    let sz1 = al1.length;
    let sz2 = al2.length;
    let i = sz1 - 1, j = 0;
    let al3 = [];
    for(; (j < sz2) && (i >= 0); i--){
        if(al1[i] > al2[j]){
            al3.push(i + 2);
            i++;
            j++;
        }
        else if(al1[i] === al2[j]){
            al3.push(i + 1);
            i++;
            j++;
        }
    }
    while(j++ < sz2){
        al3.push(1);
    }
    return al3;
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const scoresCount = parseInt(readLine(), 10);

    const scores = readLine().split(' ').map(scoresTemp => parseInt(scoresTemp, 10));

    const aliceCount = parseInt(readLine(), 10);

    const alice = readLine().split(' ').map(aliceTemp => parseInt(aliceTemp, 10));

    let result = climbingLeaderboard(scores, alice);

    ws.write(result.join("\n") + "\n");

    ws.end();
}

File: Implementation/ClimbingTheLeaderboard.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    // Complete the climbingLeaderboard function below.
    static ArrayList<Integer> climbingLeaderboard(ArrayList<Integer> al1, ArrayList<Integer> al2) {
        int sz1 = al1.size();
        int sz2 = al2.size();
        int i = sz1 - 1, j = 0;
        ArrayList<Integer> al3 = new ArrayList<Integer>();
        for(; (j < sz2) && (i >= 0); i--){
            if(al1.get(i) > al2.get(j)){
                al3.add(i + 2);
                i++;
                j++;
            }
            else if(al1.get(i) == al2.get(j)){
                al3.add(i + 1);
                i++;
                j++;
            }
        }
        while(j++ < sz2){
            al3.add(1);
        }
        return al3;
    }
    private static final Scanner sc = new Scanner(System.in);

    public static void main(String[] args) {
        int scoresCount = sc.nextInt();
        TreeSet<Integer> ts = new TreeSet<Integer>();
        for (int i = 0; i < scoresCount; i++) {
            int scoresItem = sc.nextInt();
            ts.add(scoresItem);
        }
        ArrayList<Integer> al1 = new ArrayList<Integer>(ts);
        Collections.reverse(al1);
        ArrayList<Integer> al2 = new ArrayList<Integer>();
        ArrayList<Integer> al3 = new ArrayList<Integer>();
        int aliceCount = sc.nextInt();
        for (int i = 0; i < aliceCount; i++) {
            int aliceItem = sc.nextInt();
            al2.add(aliceItem);
        }
        al3 = climbingLeaderboard(al1,al2);
        for (int i = 0; i < aliceCount; i++) {
            System.out.println(al3.get(i));
        }
        sc.close();
    }
}
